ratatui-core/src/style/stylize.rs

```diff
@@ -114,17 +112,33 @@ impl fmt::Debug for ColorDebug {
 /// }
 /// ```
 macro_rules! color {
-    ( $color:ident ) => {
-        paste! {
+    ( $vis:vis $color:ident() -> $ty:ty ) => {
+        paste::paste! {
+            #[doc = "Sets the foreground color to [`" $color "`](Color::" $color:camel ")."]
+            #[must_use = concat!("`", stringify!($color), "` returns the modified style without modifying the original")]
+            $vis fn $color(self) -> $ty {
+                self.fg(Color::[<$color:camel>])
+            }
+
+            #[doc = "Sets the background color to [`" $color "`](Color::" $color:camel ")."]
+            #[must_use = concat!("`on_", stringify!($color), "` returns the modified style without modifying the original")]
+            $vis fn [<on_ $color>](self) -> $ty {
+                self.bg(Color::[<$color:camel>])
+            }
+        }
+    };
+
+    ( $vis:vis const $color:ident() -> $ty:ty ) => {
+        paste::paste! {
             #[doc = "Sets the foreground color to [`" $color "`](Color::" $color:camel ")."]
             #[must_use = concat!("`", stringify!($color), "` returns the modified style without modifying the original")]
-            fn $color(self) -> T {
+            $vis const fn $color(self) -> $ty {
```

Comment:
Yeah, I don't think so too. We can create two macros and call the other one from here if we **really** want to avoid repetition though.

I think it's fine like this tbh.